src/hooks/useGridDimensions.ts

```diff
@@ -40,5 +38,36 @@ export function useGridDimensions(): [
     };
   }, []);
 
-  return [gridRef, inlineSize, blockSize, isWidthInitialized];
+  const observeMeasuringCell = useMemo(() => {
+    const resizeObserver = new ResizeObserver((entries) => {
+      setMeasuredColumnWidths((measuredColumnWidths) => {
+        const newMeasuredColumnWidths = new Map(measuredColumnWidths);
+        let hasChanges = false;
+
+        for (const entry of entries) {
+          const { inlineSize } = entry.contentBoxSize[0];
+          const key = (entry.target as HTMLDivElement).dataset.measuringCellKey!;
+          newMeasuredColumnWidths.set(key, inlineSize);
+          hasChanges ||= measuredColumnWidths.get(key) !== inlineSize;
+        }
+
+        return hasChanges ? newMeasuredColumnWidths : measuredColumnWidths;
+      });
+    });
+
+    return (measuringCell: HTMLDivElement | null) => {
```

Comment:
Does the `resizeObserver` need to be disconnected?

Reply:
It should be garbage-collected correctly.

https://twitter.com/jaffathecake/status/1405437361643790337
https://bugs.chromium.org/p/chromium/issues/detail?id=1220041